lib/puppetserver/ca/utils/http_client.rb

```diff
@@ -176,7 +176,7 @@ def make_store(bundle, crl_usage, crls = nil)
         # Returns true if it receives back a response of "running", and false if
         # no connection can be made, or a different response is received.
         def self.check_server_online(settings, logger)
-          status_url = URL.new('https', settings[:ca_server], settings[:ca_port], 'status', 'v1', 'simple', 'ca')
+          status_url = URL.new('https', settings[:ca_server], settings[:ca_port], 'status', 'v1', 'simple', 'ca', {})
```

Comment:
What exactly was the error you were seeing that lead to this? I think `:query` should be an optional param, so you don't need this, but we may need to update the URL constructor to check if `query` is nil, in addition to checking if it's empty.

Reply:
Since this line called `URL.new` directly instead of going through the `make_ca_url` function in `certificate_authority.rb`, the `:query` is automatically set to `nil` if you call `URL.new` without passing in a hashmap.

Reply:
right, that's why I think we should adjust the check in `full_url` to be 
```
url = url + "?" + URI.encode_www_form(query) unless (query.nil? || query.empty?)
```
(not positive if the parentheses are needed, but added for clarity)

Reply:
That is a better solution! Thanks Molly!